src/io/network/endpoint.cpp

```diff
@@ -109,4 +111,15 @@ std::ostream &operator<<(std::ostream &os, const Endpoint &endpoint) {
   return os << endpoint.address << ":" << endpoint.port;
 }
 
+bool Endpoint::ValidReplicaAddress() const {
```

Comment:
To specific, this should be replica agnostic code. Better name "IsResolvableAddress"